src/sinks/s3_common/sink.rs

```diff
@@ -61,8 +64,11 @@ where
             .request_builder(builder_limit, request_builder)
             .filter_map(|request| async move {
                 match request {
-                    Err(e) => {
-                        error!("Failed to build S3 request: {:?}.", e);
+                    Err(error) => {
+                        emit!(SinkRequestBuildError {
+                            name: S3SinkConfig::NAME, // This is also used by the `datadog_archives` sink so...
```

Comment:
It wasn't clear to me how we could nicely pass in the name for either `aws_s3` or `datadog_archives`

Reply:
Ohh. Yes that is a problem. I didn't consider/realize at the time that there were implementations used by more than one sink that would have different names ...

I wonder if the `RequestBuilder` trait could have a `name()` function... :thinking: seems a little hackish. Pondering.

Reply:
I wonder if we could we use something similar to [`tracing::Instrument`](https://docs.rs/tracing/0.1.36/tracing/trait.Instrument.html) to setup a context within which the request builder and the like are being run? Values like this could then be pulled from the context..

Reply:
Dropping this field altogether because `tracing` includes the component id already 😄